src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerLatencyBasedHealthCheck.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    public static String nameFor(String pollingUri) {
+        return ClientPollerHealthChecks.nameFor(TYPE, pollingUri);
+    }
+
+    /**
+     * Create instance using given {@link ClientPoller} and {@link PollerHealthCheckConfig}
+     *
+     * @param poller            The poller to pull the statistics from
+     * @param healthCheckConfig The configuration for the health checks
+     * @return a new instance of the health check
+     */
+    public static ClientPollerLatencyBasedHealthCheck of(ClientPoller poller, PollerHealthCheckConfig healthCheckConfig) {
+        return of(
+                poller.statistics(),
+                healthCheckConfig.getAverageLatencyWarningThreshold().toMilliseconds(),
+                ChronoUnit.MILLIS
+        );
+    }
+
+    /**
+     * Create instance using given statistics and a default latency threshold of {@link #DEFAULT_AVG_LATENCY_WARNING_THRESHOLD_MILLIS}.
+     *
+     * @param statistics The statistics to use for tracking latency health
+     * @return a new instance of the health check
+     */
+    public static ClientPollerLatencyBasedHealthCheck of(ClientPollerStatistics statistics) {
+        return of(statistics, DEFAULT_AVG_LATENCY_WARNING_THRESHOLD_MILLIS, DEFAULT_AVG_LATENCY_WARNING_THRESHOLD_UNIT);
+    }
+
+    /**
+     * Create instance using given statistics, and specified latency warning threshold.
+     *
+     * @param statistics                     The statistics to use for tracking latency health
+     * @param avgLatencyWarningThreshold     The warning threshold for latency
+     * @param avgLatencyWarningThresholdUnit The units that correspond to the avgLatencyWarningThresold
+     * @return a new instance of the health check
+     */
+    public static ClientPollerLatencyBasedHealthCheck of(ClientPollerStatistics statistics,
+                                                         long avgLatencyWarningThreshold,
+                                                         TemporalUnit avgLatencyWarningThresholdUnit) {
+        return new ClientPollerLatencyBasedHealthCheck(statistics, avgLatencyWarningThreshold, avgLatencyWarningThresholdUnit);
+    }
+
+    @Override
+    protected Result check() {
+        var avgPollLatencyInMillis = statistics.averagePollLatencyInMillis();
+
+        if (avgPollLatencyInMillis < avgLatencyWarningThresholdMillis) {
+            return newHealthyResult("Poller average latency: %.2f millis", avgPollLatencyInMillis);
+        }
+
+        return ClientPollerHealthChecks.unhealthy(
+                statistics,
+                "Poller average latency %.2f millis is %s warning threshold (%.2f)",
+                avgPollLatencyInMillis,
+                howMuchAbove(avgPollLatencyInMillis),
+                avgLatencyWarningThresholdMillis
+        );
+    }
+
+    private String howMuchAbove(double avgPollLatencyInMillis) {
+        if (avgPollLatencyInMillis - avgLatencyWarningThresholdMillis < 1E-2) {
```

Comment:
Huh, 0.01 ... should we change the "barely above" definition to something higher? Like 0.1 or 0.5, 1, or even higher like 50 or 100 especially since our threshold is 3,500?

Reply:
All this does is add the word "barely" to the failure message.  I think you made it this value because we strip the number down to 2 decimal places in the output and it looked weird as to why the number matched the threshold.